### PR TITLE
Fix for issue 7129: Automatic keyword groups do not update in group list when keywords change

### PR DESCRIPTION
**Pull request by Gan-Cheng**
Fixes https://github.com/JabRef/jabref/issues/7129

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

### Bug:
It is possible to create dynamic, auto-generated groups based on the contents of a field, e.g. keywords. (In the Groups interface, Add subgroup, and choose `Collect by > Specified keywords`). However, these dynamically generated groups don't update when new keywords appear in the file. In other words, if you edit an item in such a way that a new keyword appears, it does not appear in the dynamic list of groups, unless you close and re-open the file or go to other groups and back.

### The way to fix:
Inspired by https://github.com/JabRef/jabref/pull/7153, I register a listener to listen for the fieldchange. What makes it different is that I add some trigger condition of updating group list: **Only when user changes value of keywords**. Besides, to scale to huge database, I make the group list only updates special group like AutomaticKeywordGroup, instead of reconstrucing the whole group tree.

### CODE REVIEW

**Review by k3KAW8Pnf7mkmdSMPHz27 — CHANGES_REQUESTED** [review]
Thank you for working on this issue!
I am not convinced this fixes enough of the performance issues, and I am not sure how to approach it. Have you looked at if `Keyword.equals` can be used to not recreate subgroups that already exist?

File: src/main/java/org/jabref/gui/StateManager.java [hunk 1]
```diff
@@ -159,4 +161,11 @@ public DialogWindowState getDialogWindowState(String className) {
     public void setDialogWindowState(String className, DialogWindowState state) {
         dialogWindowStates.put(className, state);
     }
+
+    /**
+     * Return the property of fieldkeywordchangeed.
+     */
```
Comment (calixtus) [inline src/main/java/org/jabref/gui/StateManager.java:167]: Please provide proper JavaDoc. This comment is completely superfluous.

File: src/main/java/org/jabref/gui/groups/GroupTreeViewModel.java [hunk 2]
```diff
@@ -59,6 +60,8 @@ public GroupTreeViewModel(StateManager stateManager, DialogService dialogService
         // Register listener
         EasyBind.subscribe(stateManager.activeDatabaseProperty(), this::onActiveDatabaseChanged);
         EasyBind.subscribe(selectedGroups, this::onSelectedGroupChanged);
+        // EasyBind.subscribe(stateManager.fieldKeywordChangedProperty(), changeEvent -> onActiveDatabaseChanged(stateManager.getActiveDatabase()));
```
Comment (calixtus) [inline src/main/java/org/jabref/gui/groups/GroupTreeViewModel.java:63]: Artifact?

**Review by calixtus — CHANGES_REQUESTED** [review]

**Comment by koppor** [comment]
Closing this issue due to inactivity :zzz: 

In case you resume work on this, please ping us here. The main comment to be adressed ist https://github.com/JabRef/jabref/pull/7753#discussion_r638853140.

**Pull request closed without merging**